# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/TextLeafRange.h
# Commit: 388a15605114
# Full Hash: 388a156051146d9c128adc05365f9c1e0aa94cb2
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-05-27 15:58:57
# Regressor Bug: 1737919
# File Overlap Count: 1
# Description:
#   Bug 1737919 part 3: Support spelling errors for cached RemoteAccessibles. r=morgan
#   
#   Even though spelling errors can cross Accessibles and are represented in the DOM as ranges, we cache them for each text leaf.
#   This is necessary so that we correctly update the offsets when text is inserted or removed from a leaf.
#   We cache them as an array of offsets, including both the start and end offset for each range.
# ==============================================================================

diff -r 08d033b9651b -r 388a15605114 accessible/base/TextLeafRange.h
--- a/accessible/base/TextLeafRange.h	Fri May 27 10:56:40 2022 +0000
+++ b/accessible/base/TextLeafRange.h	Fri May 27 10:56:41 2022 +0000
@@ -15,6 +15,7 @@
 
 namespace mozilla::a11y {
 class Accessible;
+class LocalAccessible;
 
 /**
  * Represents a point within accessible text.
@@ -129,6 +130,13 @@
       bool aIncludeDefaults = true) const;
 
   /**
+   * Get the offsets of all spelling errors in a given LocalAccessible. This
+   * should only be used when pushing the cache. Most callers will want
+   * FindTextAttrsStart instead.
+   */
+  static nsTArray<int32_t> GetSpellingErrorOffsets(LocalAccessible* aAcc);
+
+  /**
    * Find the start of a run of text attributes in a specific direction.
    * A text attributes run is a span of text where the attributes are the same.
    * If no boundary is found, the start/end of the container is returned